File: schemas/item_cardapio.py
from pydantic import BaseModel
from typing import Optional


class ItemCardapioCreate(BaseModel):
    idRestauranteItemc: str
    dscNomeItemc: str
    dscInformacaoItemc: Optional[str] = None
    valPrecoItemc: float
    dscDisponibilidadeItemc: str


class ItemCardapioResponse(BaseModel):
    idItemc: str
    idRestauranteItemc: str
    dscNomeItemc: str
    dscInformacaoItemc: Optional[str] = None
    valPrecoItemc: float
    dscDisponibilidadeItemc: str

File: frontend/src/types/itemCardapio.ts
import { Endereco } from './endereco';

export interface ItemCardapio {
  idItemc: string;
  idRestauranteItemc: string;
  dscNomeItemc: string;
  dscInformacaoItemc?: string;
  valPrecoItemc: number;
  dscDisponibilidadeItemc: string;
}

export interface ItemCardapioCreate {
  idRestauranteItemc: string;
  dscNomeItemc: string;
  dscInformacaoItemc?: string;
  valPrecoItemc: number;
  dscDisponibilidadeItemc: string;
}